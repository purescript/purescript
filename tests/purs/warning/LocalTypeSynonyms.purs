-- @shouldWarnWith UnusedName
module Main where

data Proxy :: forall k. k -> Type
data Proxy a = Proxy

generalized :: forall f. Proxy f -> Int
generalized _ = 0
  where
  type LocalApp a = f a -- This is being tested not to warn about polymorphic kinds
