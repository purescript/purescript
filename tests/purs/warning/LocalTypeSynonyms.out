Warning found:
in module [33mMain[0m
at tests/purs/warning/LocalTypeSynonyms.purs:10:3 - 10:24 (line 10, column 3 - line 10, column 24)

  The type [33mLocalApp[0m was introduced but not used.

in value declaration [33mgeneralized[0m

See https://github.com/purescript/documentation/blob/master/errors/UnusedName.md for more information,
or to contribute content related to this warning.

